tmt/steps/provision/mrack.py

```diff
@@ -1123,13 +1143,39 @@ async def __init__(self, guest: 'GuestBeaker') -> None:  # type: ignore[misc]
             raise ProvisionError(
                 f"Configuration file missing: {missing_conf_err.filename}"
             ) from missing_conf_err
+        except Exception as e:
+            raise ProvisionError(f"Failed to initialize mrack transformer: {e}")
 
         self._mrack_provider = self._mrack_transformer._provider
         self._mrack_provider.poll_sleep = DEFAULT_PROVISION_TICK
 
         if guest.job_id:
             self._bkr_job_id = guest.job_id
 
+    def _create_provider_config(self, config_data: dict[str, Any]) -> dict[str, Any]:
+        """
+        Create provider configuration from parsed YAML config (thread-safe)
+        """
+        try:
+            # The YAML config should already be in the correct format for mrack
+            # Just validate that it has the necessary structure
+            if not isinstance(config_data, dict):
+                raise ProvisionError(
+                    "Invalid mrack provisioning configuration format - expected dictionary"
+                )
+
+            # Check for required provider configurations
+            if 'beaker' not in config_data:
+                raise ProvisionError(
+                    "No 'beaker' section found in mrack provisioning configuration"
+                )
+
+            # Return the configuration data as-is since it's already in the correct format
+            return config_data
+
+        except Exception as e:
+            raise ProvisionError(f"Failed to parse mrack provisioning configuration: {e}")
```

Comment:
`raise ... from e` instead of embedding `e` into the message. It preserves the sequence of exceptions, with their tracebacks, and it is reported correctly by tmt.